js/style/style_layer.js

```diff
@@ -231,7 +231,9 @@ StyleLayer.prototype = util.inherit(Evented, {
     updatePaintTransition: function(name, classes, options, globalOptions, animationLoop) {
         var declaration = this._paintDeclarations[''][name];
         for (var i = 0; i < classes.length; i++) {
-            declaration = this._paintDeclarations[classes[i]][name] || declaration;
+            if (this._paintDeclarations[classes[i]] && this._paintDeclarations[classes[i]][name]) {
+                declaration = this._paintDeclarations[classes[i]][name];
```

Comment:
Now that `this._paintDeclarations[classes[i]]` repeats 3 times, let's put it in a variable.